### PR TITLE
Nonlinear sdi

### PR DESCRIPTION
**Pull request by sungeunbae**
On request from @Vahid-Loghman  and @lee-robin .

rotd_values are now obtained by rotating first, then Burks_Baker_2013. Previously it computed the displacements from Burks_Baker_2013 first, then rotated them. 

@Vahid-Loghman verified the output from this code.

### CODE REVIEW

**Review by jasonmotha — DISMISSED** [review]

File: IM_calculation/IM/intensity_measures.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def get_SDI(acceleration, period, DT, z, alpha, dy, dt):
     ).T
 
 
-def get_SDI_nd(acceleration, period, NT, DT, z, alpha, dy, dt):
```
Comment (jpa198) [inline IM_calculation/IM/intensity_measures.py:56]: Good catch

File: IM_calculation/Advanced_IM/Models/Burks_Baker_2013/run.py [hunk 2]
```diff
@@ -88,9 +95,10 @@ def main(comp_000: Path, comp_090: Path, rotd: bool, output_dir: Path):
                             results[component.str_value][im_name] = sdi_values[j, i]
                     if not rotd:
                         continue
-                    rotd_values = im_calculation.calculate_rotd(
-                        displacements, rotd_comps
-                    )
+                    # computing non-linear rotd
+                    func = lambda x: np.max(np.abs(intensity_measures.get_SDI_nd(x, period, DT, z, alpha, dy, dt) * 100), axis=1)
```
Comment (jpa198) [inline IM_calculation/Advanced_IM/Models/Burks_Baker_2013/run.py:99]: Nice

File: IM_calculation/Advanced_IM/Models/Burks_Baker_2013/run.py [hunk 3]
```diff
@@ -60,18 +62,23 @@ def main(comp_000: Path, comp_090: Path, rotd: bool, output_dir: Path):
 
     accelerations = np.array((waveforms["000"], waveforms["090"])).T
     ordered_columns_dict = {}
+    sd_rotd_list = []
```
Comment (jpa198) [inline IM_calculation/Advanced_IM/Models/Burks_Baker_2013/run.py:65]: Not needed with the lambda

File: IM_calculation/Advanced_IM/Models/Burks_Baker_2013/run.py [hunk 4]
```diff
@@ -60,18 +62,23 @@ def main(comp_000: Path, comp_090: Path, rotd: bool, output_dir: Path):
...
+
 
     for z in z_list:
         for dt in dt_list:
             for alpha in alpha_list:
                 for dy in dy_list:
```
Comment (jpa198) [inline IM_calculation/Advanced_IM/Models/Burks_Baker_2013/run.py:71]: TODO: replace the quad for loop with itertools.product (Would reduce the indentation by 3 levels)
Not needed for this PR, could add in the TODO comment though

**Review by jpa198 — CHANGES_REQUESTED** [review]
Just need to drop sd_rotd_list then good to go

File: IM_calculation/Advanced_IM/Models/Burks_Baker_2013/run.py [hunk 5]
```diff
@@ -24,10 +24,12 @@
 STORIES = 10
 
 z_list = [0.05]  # damping ratio
-alpha_list = [0.05]  # strain hardening ratio
-dy_list = [0.1765, 0.4, 0.6]  # strain hardening ratio
+alpha_list = [0.0001]  # strain hardening ratio
+dy_list = [0.21278]  # strain hardening ratio
 dt_list = [0.005]
-period = np.array(constants.DEFAULT_PSA_PERIODS)
+g = 9.81
```
Comment (jpa198) [inline IM_calculation/Advanced_IM/Models/Burks_Baker_2013/run.py:30]: Does this get used?

**Review by jasonmotha — APPROVED** [review]
Good catch. I remember seeing it added but couldn't remember where it ended up.

**Review by jpa198 — APPROVED** [review]

**Pull request merged**